flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/json/JobExecutionResultDeserializer.java

```diff
@@ -0,0 +1,153 @@
+ */
+
+package org.apache.flink.runtime.rest.messages.json;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.jobmaster.JobExecutionResult;
+import org.apache.flink.util.SerializedThrowable;
+import org.apache.flink.util.SerializedValue;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.JsonParser;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.JsonToken;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.DeserializationContext;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.JavaType;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.type.TypeFactory;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkState;
+
+/**
+ * JSON deserializer for {@link JobExecutionResult}.
+ *
+ * @see JobExecutionResultSerializer
+ */
+public class JobExecutionResultDeserializer extends StdDeserializer<JobExecutionResult> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final JobIDDeserializer jobIdDeserializer = new JobIDDeserializer();
+
+	private final SerializedThrowableDeserializer serializedThrowableDeserializer =
+		new SerializedThrowableDeserializer();
+
+	private final SerializedValueDeserializer serializedValueDeserializer;
+
+	public JobExecutionResultDeserializer() {
+		super(JobExecutionResult.class);
+		final JavaType objectSerializedValueType = TypeFactory.defaultInstance()
+			.constructType(new TypeReference<SerializedValue<Object>>() {
+			});
+		serializedValueDeserializer = new SerializedValueDeserializer(objectSerializedValueType);
+	}
+
+	@Override
+	public JobExecutionResult deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
+		JobID jobId = null;
+		long netRuntime = -1;
+		SerializedThrowable serializedThrowable = null;
+		Map<String, SerializedValue<Object>> accumulatorResults = null;
+
+		while (true) {
+			final JsonToken jsonToken = p.nextToken();
+			assertNotEndOfInput(p, jsonToken);
+			if (jsonToken == JsonToken.END_OBJECT) {
+				break;
+			}
+
+			final String fieldName = p.getValueAsString();
+			switch (fieldName) {
+				case JobExecutionResultSerializer.FIELD_NAME_JOB_ID:
+					assertNextToken(p, JsonToken.VALUE_STRING);
+					jobId = jobIdDeserializer.deserialize(p, ctxt);
+					break;
+				case JobExecutionResultSerializer.FIELD_NAME_NET_RUNTIME:
+					assertNextToken(p, JsonToken.VALUE_NUMBER_INT);
+					netRuntime = p.getLongValue();
+					break;
+				case JobExecutionResultSerializer.FIELD_NAME_ACCUMULATOR_RESULTS:
+					assertNextToken(p, JsonToken.START_OBJECT);
+					accumulatorResults = parseAccumulatorResults(p, ctxt);
```

Comment:
Actually now I would prefer to register the `SerializedThrowableDeserializer` as a Jackson module in the `ObjectMapper`. This is ugly.